fineract-provider/src/main/java/org/apache/fineract/portfolio/loanproduct/domain/LoanProduct.java

```diff
@@ -102,6 +103,10 @@
     @JoinTable(name = "m_product_loan_charge", joinColumns = @JoinColumn(name = "product_loan_id"), inverseJoinColumns = @JoinColumn(name = "charge_id"))
     private List<Charge> charges;
 
+    @ManyToMany(fetch = FetchType.EAGER)
+    @JoinTable(name = "m_product_loan_rate", joinColumns = @JoinColumn(name = "product_loan_id"), inverseJoinColumns = @JoinColumn(name = "rate_id"))
```

Comment:
@ManyToMany or OneToMany
Can the functionality be achieved by using fetch type lazy

Reply:
It should be ManyToMany as we use a third table to join the rate and loan_product tables in the same way that the charges and loan_products are joined.  About the fetch type Lazy I guess it should work the same but I will do some tests to be sure.